frontend/src/metabase/components/DownloadButton.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { Component, PropTypes } from "react";
+import ReactDOM from "react-dom";
+
+import Button from "metabase/components/Button.jsx";
+
+export default class DownloadButton extends Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {};
+    }
+
+    static propTypes = {
+        url: PropTypes.string.isRequired,
+        method: PropTypes.string,
+        params: PropTypes.object,
+        icon: PropTypes.string,
+    };
+
+    static defaultProps = {
+        icon: "downarrow",
+        method: "POST",
+        params: {}
+    };
+
+    render() {
+        const { children, url, method, params, ...props } = this.props;
+        return (
+            <form ref={(c) => this._form = c} method={method} action={url}>
+                { Object.entries(params).map(([name, value]) =>
+                    <input key={name} type="hidden" name={name} value={value} />
+                )}
+                <Button
+                    onClick={() => ReactDOM.findDOMNode(this._form).submit()}
```

Comment:
Is there a way to handle the submission via component state here or is the form required because of the fact that we're downloading a file? If we're not able to use the component state I'd just as soon remove it from up above in the constructor so it's clear the component has no local state. 

(It's too bad you can't use refs with stateless functional components).